lucille-core/src/main/java/com/kmwllc/lucille/core/Indexer.java

```diff
@@ -48,12 +51,38 @@ public void terminate() {
 
   public Indexer(Config config, IndexerMessageManager manager, String metricsPrefix) {
     this.manager = manager;
-    this.idOverrideField = config.hasPath("indexer.idOverrideField") ? config.getString("indexer.idOverrideField") : null;
-    this.indexOverrideField = config.hasPath("indexer.indexOverrideField") ? config.getString("indexer.indexOverrideField") : null;
-    this.ignoreFields = config.hasPath("indexer.ignoreFields") ? config.getStringList("indexer.ignoreFields") : null;
-    int batchSize = config.hasPath("indexer.batchSize") ? config.getInt("indexer.batchSize") : DEFAULT_BATCH_SIZE;
-    int batchTimeout = config.hasPath("indexer.batchTimeout") ? config.getInt("indexer.batchTimeout") : DEFAULT_BATCH_TIMEOUT;
-    this.batch = (indexOverrideField == null) ? new SingleBatch(batchSize, batchTimeout) : new MultiBatch(batchSize, batchTimeout, indexOverrideField);
+    this.idOverrideField =
+        config.hasPath("indexer.idOverrideField")
+            ? config.getString("indexer.idOverrideField")
+            : null;
+    this.indexOverrideField =
+        config.hasPath("indexer.indexOverrideField")
+            ? config.getString("indexer.indexOverrideField")
+            : null;
+    this.ignoreFields =
+        config.hasPath("indexer.ignoreFields")
+            ? config.getStringList("indexer.ignoreFields")
+            : null;
+    this.deletionMarkerField =
+        config.hasPath("indexer.deletionMarkerField")
+            ? config.getString("indexer.deletionMarkerField")
+            : null;
+    this.deletionMarkerFieldValue =
+        config.hasPath("indexer.deletionMarkerFieldValue")
+            ? config.getString("indexer.deletionMarkerFieldValue")
+            : null;
```

Comment:
if people are going to use "isDelete" or "delete" as the marker field most of the time, it might be nice to set a default of "true" for the deletionMarkerFieldValue, but I guess we can't be sure of that. Thinking out loud -- no action needed.